rfcs/40-unify-object-metadata.md

```diff
@@ -52,23 +52,44 @@ So, I propose to split all metadata into four groups which based on the definer:
     - ...
 - user metadata: defined via user input.
 
-Both `global` and `standard` metadata will be included in object type. Both `service` and `user-defined` metadata will be stored in a hash map inside object type.
+Both `global` and `standard` metadata will be included in object type.
+`service` metadata will be stored in a struct that defined by service.
+`user-defined` metadata will be stored in a hash map inside object type.
 
 Use `golang` as example:
 
 - `id` will be `id string`
 - `content md5` will be `contentMd5 string`
-- service metadata will be `serviceMetadata map[string]interface{}`
+- service metadata will be `serviceMetadata interface{}`
 - user metadata will be `userMetadata map[string]string`
 
+For service metadata,  we will introduce `Strong Typed Service Metadata`. We can generate following struct:
+
+```go
+type ObjectMetadata struct {
```

Comment:
Maybe we can add two different service metadata as example, to show the difference and compatiblity.